### ISSUE TITLE
Add support for multiple timesteps in Noise Schedulers

### ISSUE THREAD

**Issue by bonlime** [huggingface/diffusers#1784]
**Is your feature request related to a problem? Please describe.**
When trying to call scheduler.step with multiple timesteps it throws an error: `Boolean value of Tensor with more than one value is ambiguous`. The origin of this error for DDIM as example is here: https://github.com/huggingface/diffusers/blob/261a448c6aa467810545d87499f6f3bed334754f/src/diffusers/schedulers/scheduling_ddim.py#L270

The reason why I need this functionality is because I'm using `scheduler.step` during training, not only inference, and different examples in the batch come from different timesteps. 

**Describe the solution you'd like**
Changing the signature of all `scheduler.step` functions `timestep: int` -> `timestep: Union[int, Tensor]` would allow performing steps with different timesteps. The change would be small, straightforward and 100% backwards compatible. 

If this change is approved, I could open the PR for it myself

**Comment by patrickvonplaten** [comment]
Hey @bonlime,

Thanks for the issue!
Why would you call `step` with multiple time values? `step` should only be used for inference in which case, we don't need multiple time values, no?

**Comment by bonlime** [comment]
Okay, seems no ones else really needs this. And since then i've been able to hack it around, so can close.

**Comment by chang-1** [comment]
actually, that is a good feature.
